File: meal_hack_flask/models/meal.py
from mongoengine import Document, StringField, FloatField, DateField, IntField, ReferenceField
from models.recipe import Recipe

class Meal(Document):
    """
    Template for mongoengine document, which represent's a meal
    :param name: required string value
    :param description: optional string value, max 240 characters
    :param price: optional float value
    :param image_url: optional string image url
    :Example:
    """

    # user_id = StringField(required=True)
    date = DateField(required=True)
    quantity = IntField(min_value=1)
    note = StringField(max_length=240)
    recipe_id = IntField(required=True)
    recipe_title = StringField(requried=True)

File: meal_hack_flutter/lib/models/meal.dart
class Meal {
  DateTime date;
  int quantity;
  String note;
  int recipeId;
  String recipeName;

  Meal({this.date, this.quantity, this.note, this.recipeId, this.recipeName});

  factory Meal.fromJson(Map<String, dynamic> json) {
    return Meal(
        date: DateTime.now(),
        quantity: json["quantity"],
        note: json["note"],
        recipeId: json["recipe_id"],
        recipeName: json["recipe_title"]);
  }
}
